src/lib/bootloader/grub2base.rb

```diff
@@ -64,14 +64,14 @@ def pmbr_setup(*devices)
       return if @pmbr_action == :nothing
 
       action_parted = case @pmbr_action
-      when :add    then "on"
-      when :remove then "off"
+      when :add    then true
+      when :remove then false
       else raise "invalid action #{action}"
       end
 
-      devices.each do |dev|
-        Yast::Execute.locally("parted", "-s", dev, "disk_set", "pmbr_boot", action_parted)
-      end
+      staging = Y2Storage::StorageManager.instance.y2storage_staging
+      disks = staging.disks.select { |d| devices.include?(d.name) }
+      disks.each { |d| d.pmbr_boot = action_parted }
```

Comment:
Oh, well, if this happens during installation before preparing the disks. Then it's fine. The `inst_prep_disk` client will commit the changes when the user press "install" in the summary screen.

But when running the bootloader module standalone or when doing changes at the end of installation, after installing the packages and all that, I guess you have to commit the changes (I'm not aware of any client doing a final commit).